core/src/main/kotlin/org/evomaster/core/problem/util/ParamUtil.kt

```diff
@@ -0,0 +1,671 @@
+        private fun suggestBindSequence(geneA: Gene, geneB: Gene) : Pair<Boolean?, Pair<Gene, Gene>>{
+            val pA = getGeneTypePriority(geneA)
+            val pB = getGeneTypePriority(geneB)
+
+            return Pair(
+                    if(pA != -1 && pB!= -1)(pA != pB) else null,
+                    if(pA > pB) Pair(geneA, geneB) else Pair(geneB, geneA)
+            )
+
+        }
+
+        /**
+         * [geneA] copy values from [geneB]
+         * @return null cannot find its priority
+         *         true keep current sequence
+         *         false change current sequence
+         */
+        private fun checkBindSequence(geneA: Gene, geneB: Gene) : Boolean?{
+            val pA = getGeneTypePriority(geneA)
+            val pB = getGeneTypePriority(geneB)
+
+            if(pA == -1 || pB == -1) return null
+
+            if(pA >= pB) return true
+
+            return false
+        }
+
+        /**
+         * @param existingData whether the data represents existing data
+         * @param enableFlexibleBind whether to enable flexible bind, which can only be enabled when [existingData] is false
+         */
+        fun bindParamWithDbAction(dbgene: Gene, paramGene: Gene, existingData: Boolean, enableFlexibleBind : Boolean = true){
+            if(dbgene is SqlPrimaryKeyGene || dbgene is SqlForeignKeyGene || dbgene is SqlAutoIncrementGene){
+                /*
+                    if gene of dbaction is PK, FK or AutoIncrementGene,
+                        bind gene of Param according to the gene from dbaction
+                 */
+                copyGene(b=getValueGene(dbgene), g=getValueGene(paramGene), b2g=false)
+            }else{
+                val db2Action = !existingData && (!enableFlexibleBind || checkBindSequence(getValueGene(dbgene), getValueGene(paramGene))?:true)
+                copyGene(b=getValueGene(dbgene), g=getValueGene(paramGene), b2g=db2Action)
+//                if(existingData){
+//                    /*
+//                     If the data is existing in db, bind gene of Param according to the gene from dbaction
+//                        otherwise, bind gene of action according to rest action, i.e., Gene of Param
+//                    */
+//                    copyGene(b=getValueGene(dbgene!!), g=getValueGene(paramGene), b2g=!existingData)
+//                }else{
+//                    if (enableFlexibleBind){
+//
+//                    }
+//                }
+
+            }
+
+        }
+
+        fun bindParam(dbAction: DbAction, param : Param, previousToken: String, existingData : Boolean) : Boolean{
+            var gene  : Gene? = null
+            var similarity = 0.0
+            dbAction.seeGenes().forEach findGene@{
+                val score = compareDBGene(dbAction, it, param.name, previousToken)
+                if(score>similarity){
+                    similarity = score
+                    gene = it
+                }
+                if(score == 1.0) return@findGene
```

Comment:
does the value `1.0` have a special meaning? if so, it could be put in a constant val, with a meaningful name, as it is used in quite a few places in this class

Reply:
this function is not used anymore. ResourceDepManageService handles the identification of params to bind the same values regarding inserted/selected row. Regarding this kind of improper implementation in other places, I tried to fix them.  Besides, unused imports and functions are also fixed.